### ISSUE TITLE
Reenable Sentry

### ISSUE THREAD

**Issue by markmhendrickson** [blockstack/blockstack-app#2822]
We used to have Sentry to collect useful data about error rates and performance in production, but it was never configured adequately enough to derive useful insights and we ended up removing it temporarily as a precautionary measure (see https://github.com/hirosystems/stacks-wallet-web/pull/2612).

We should reenable it but this time with proper configuration. 

@kyranjamie can provide some historical context on why we integrate this directly rather than using Segment in case it's relevant for the reenabling process

**Comment by kyranjamie** [comment]
Segment's an integration mgmt tool making it easy to forward analytics events to many tools. It supports more complex use cases, such as error catching, but these use cases are less well supported:
- Documentation is sparse, incomplete, or missing
- Sentry documentation focuses on using Sentry tooling directly, not via Segment
- Sentry configuration via Segment is limited, or outdated
- `Error -> Sentry -> Segment` is more complex to set up than `Error -> Sentry`

**Comment by edu-stx** [comment]
As part of #2791, a `settings` slice was added to the store to record user settings, such as a user's analytics consent response implemented in #2793. The location of these settings (the store in the app context) presents a challenge in getting this information to `shared/` code that may be used by various contexts, and where the Sentry config lives.

I’ve got a working implementation, although there are, it seems, some trade-offs we have to think about (I wasn’t able to find a solution that didn’t involve some form of trade-off).

The `redux-persist` package has a `getStoredState` that works just fine. The issue is that it needs access to the same configuration used to set it up. Currently, this configuration lives in `app/store/index.ts`, and due to our `dependency-cruiser` rules, it can’t be imported into `shared/*`, which is where the `initSentry` function lives.

A quick workaround is to move the persistence configuration into `shared` and then allow the `initSentry` function and `app/store/index.ts` to read it from there. Up to here, seems acceptable.

The more complex issue then becomes the types, and unless we find a better solution, here is where the trade offs start. The `getStoredState` returns `Promise<object | undefined>`, so its return type needs to be coerced if we want types. The objective here is reading `state.settings.hasAllowedAnalytics` in the most reliable way possible.

**Option 1** is to create an ad-hoc type for the store. This type would need to have all props marked as optional, because it is decoupled from the actual store’s type, meaning consumers have to check every?.single?.prop to ensure it’s there. If the store’s type changes and the consent setting is moved, the Sentry code won’t be functional (it will be fail-safe and disabled), but won’t throw any errors.

**Option 2** is allowing access to the state’s type by modifying dep-cruiser. E.g., it can be moved out to a single file within `app/*`, and only that file can be made available to the rest of the app. This would allow to read the state safely, although manually, since allowing exporting selectors would require us to modify dep-cruiser further.

---

@kyranjamie 's suggestion to make use of `dependencyTypesNot`, allowing us to go for a similar solution to option 2,

```js
    {
      name: 'script-context-not-to-another',
      comment: 'One script context must not depend on another',
      severity: 'error',
      from: { path: '(^src/)([^/]+)/' },
      to: {
        path: '^$1',
        pathNot: ['$1$2', '^src/shared'],
        dependencyTypesNot: ['type-only'], // allows `import type` statements
      },
    },
```
